### ISSUE TITLE
How to seperate the skeletons of each person?

### ISSUE THREAD

**Issue by Lotayou** [tensorboy/pytorch_Realtime_Multi-Person_Pose_Estimation#36]
@tensorboy @lyxok1 
Hi, I just ran the picture_demo using the default image `ski.jpg` provided in your repo and found something wrong: As you can see, there are only 5 people in the image, so I'm expecting the estimated poses to be stored in a 18 * 5 * 4 matrix. However, this is what I've got in `all_peaks`:

![image](https://user-images.githubusercontent.com/33449901/43054851-1c91b69e-8e66-11e8-93fb-56655ec8030b.png)

I have two questions:
1. Why does line 04 and line 12 has 6 elements? It seems that the estimator has detected 6 entities out of an image that only contains 5 people. What do you make of that?

2. I don't know how to assign keypoints of each joint to the corresponding persons. For example, line 17 only contains 4 keypoints, should I assign them to person 0,1,2,3 or 0,1,2,4, ... or 1,2,3,4? Can you tell me in which variable is the assignment information stored?

Thanks for your answer!

**Comment by Lotayou** [comment]
@tensorboy any comments?

**Comment by tensorboy** [comment]
It's a little bit harder for me to figure it out, I'm still thinking about it.

**Comment by pinkfloyd06** [comment]
Hi @tensorboy & @Lotayou ,

1) I'm confronted to the same problem. How did you solve that ? any cue ?


2)Let's suppose that heatmaps containt exactly five person. Are the key points ordered by person  ?

```
all_peaks[0]=[(615, 40, 0.7343657, 0), (299, 74, 0.8005683, 1), (478, 94, 0.72995293, 2), (63, 268, 0.7753145, 3), (169, 279, 0.79473287, 4)]

all_peaks[1]=[(613, 79, 0.7158987, 5), (302, 109, 0.7536149, 6), (456, 140, 0.65013695, 7), (61, 301, 0.7433309, 8), (166, 304, 0.77520543, 9)]
```
Does 
all_peaks[0][0],all_peaks[1][0],..., all_peaks[17][0] corresponds  to the  graph of first person.

all_peaks[0][1],all_peaks[1][1],..., all_peaks[17][1] corresponds  to the  graph of second person.

all_peaks[0][4],all_peaks[1][4],..., all_peaks[17][4] corresponds  to the  graph of fifth person. ?

Thank you a lot for your answer

**Comment by Lotayou** [comment]
@pinkfloyd06 Yeah I think ideally it SHOULD be like that, but there seems to be some ambiguity in the way the results are organized, and I still don't have a conclusive answer. 

I've already switched to the Windows Portable Demo instead since I mostly code with my laptop. However, it seems the demo also supports Linux/Mac system as well. So if you are just looking for a tool to extract pose features with I suggest you use Openpose C++ .exe instead.

BTW, The Dark Side of the Moon is a classic:)

**Comment by wwdok** [comment]
Do you have any good solution now ？
